src/main/java/core/basesyntax/strategy/PurchaseHandler.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.FruitTransaction;
+import java.util.List;
+
+public class PurchaseHandler implements OperationHandler {
+    private static final String PURCHASE = "p";
+
+    @Override
+    public List<FruitTransaction> getFruitFinalQuantity(List<FruitTransaction> listOfAll,
+                                                        List<FruitTransaction> listOfFruits) {
+        for (FruitTransaction listOfFruit : listOfFruits) {
+            int oldQuantity = 0;
+            for (FruitTransaction fruitTransaction : listOfAll) {
+                if (listOfFruit.getFruit().equals(fruitTransaction.getFruit())
+                        && fruitTransaction.getType().equals(PURCHASE)) {
+                    int quantity = fruitTransaction.getQuantity();
+                    listOfFruit.setQuantity(quantity + oldQuantity);
+                    oldQuantity = quantity;
+                }
+            }
+        }
+        return listOfFruits;
```

Comment:
The logic in 'getFruitFinalQuantity' seems flawed. It's supposed to handle the purchase operation, but it's adding quantities instead of subtracting from the available quantity. Additionally, the check for the 'PURCHASE' type in line 16 is unnecessary since this handler should only deal with purchase operations. Also, there is no check for a negative result balance which should throw a RuntimeException according to the checklist.